system/Router/RouteCollection.php

```diff
@@ -992,6 +992,19 @@ public function cli(string $from, $to, ?array $options = null): RouteCollectionI
         return $this;
     }
 
+    /**
+     * Specifies a route that will only display a view.
+     * Only works for GET requests.
+     */
+    public function view(string $from, string $view, ?array $options = null): RouteCollectionInterface
+    {
+        $to = static fn (...$data) => view($view, ['page' => $data]);
```

Comment:
This adds a new dependency on the View class, which I think is acceptable but I don't like that it is hidden by the global function. I recommend changing this to use the actual class and adjust Deptrac (as necessary).

Reply:
I agree, but it is difficult to notice this because global functions hide dependencies.

Reply:
@MGatner I assume you mean use `Services::renderer()` here? If we call the class directly it won't take any custom View classes into account when rendering the view.

Reply:
@MGatner If we want to detect a new dependency, we need to code like the following:

```diff
--- a/system/Router/RouteCollection.php
+++ b/system/Router/RouteCollection.php
@@ -14,6 +14,7 @@ namespace CodeIgniter\Router;
 use Closure;
 use CodeIgniter\Autoloader\FileLocator;
 use CodeIgniter\Router\Exceptions\RouterException;
+use CodeIgniter\View\View;
 use Config\App;
 use Config\Modules;
 use Config\Services;
@@ -998,7 +999,9 @@ class RouteCollection implements RouteCollectionInterface
      */
     public function view(string $from, string $view, ?array $options = null): RouteCollectionInterface
     {
-        $to = static fn (...$data) => Services::renderer()
+        /** @var View $renderer */
+        $renderer = Services::renderer();
+        $to       = static fn (...$data) => $renderer
             ->setData(['segments' => $data], 'raw')
             ->render($view, $options);
 
```

Then we can see the errors.

```console
$ deptrac 
 507/507 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ----------- -------------------------------------------------------------------------------------------- 
  Reason      Router                                                                                      
 ----------- -------------------------------------------------------------------------------------------- 
  Violation   CodeIgniter\Router\RouteCollection must not depend on CodeIgniter\View\View (View)          
              /Users/kenji/work/codeigniter/official/CodeIgniter4/system/Router/RouteCollection.php:1002  
  Violation   CodeIgniter\Router\RouteCollection must not depend on CodeIgniter\View\View (View)          
              /Users/kenji/work/codeigniter/official/CodeIgniter4/system/Router/RouteCollection.php:17    
 ----------- -------------------------------------------------------------------------------------------- 


 -------------------- ----- 
  Report                    
 -------------------- ----- 
  Violations           2    
  Skipped violations   32   
  Uncovered            813  
  Allowed              613  
  Warnings             0    
  Errors               0    
 -------------------- -----
```

Reply:
Yeah that was my mistake, I forgot how much work `view()` was doing. This is fine, or even reverting to the helper.